sea-orm-macros/src/derives/from_query_result.rs

```diff
@@ -5,22 +5,81 @@ use syn::{
     ext::IdentExt, punctuated::Punctuated, token::Comma, Data, DataStruct, Fields, Generics, Meta,
 };
 
-pub struct FromQueryResultItem {
-    pub skip: bool,
+enum ItemType {
+    Normal,
+    Skipped,
+    Nested,
+}
+
+struct FromQueryResultItem {
+    pub typ: ItemType,
     pub ident: Ident,
 }
-impl ToTokens for FromQueryResultItem {
+
+/// Initially, we try to obtain the value for each field and check if it is an ordinary DB error
+/// (which we return immediatly), or a null error.
+///
+/// ### Background
+///
+/// Null errors do not necessarily mean that the deserialization as a whole fails,
+/// since structs embedding the current one might have wrapped the current one in an `Option`.
+/// In this case, we do not want to swallow other errors, which are very likely to actually be
+/// programming errors that should be noticed (and fixed).
+struct TryFromQueryResultCheck<'a>(&'a FromQueryResultItem);
+
+impl<'a> ToTokens for TryFromQueryResultCheck<'a> {
+    fn to_tokens(&self, tokens: &mut TokenStream) {
+        let FromQueryResultItem { ident, typ } = self.0;
+
+        match typ {
+            ItemType::Normal => {
+                let name = ident.unraw().to_string();
+                tokens.extend(quote! {
+                    let #ident = match row.try_get_nullable(pre, #name) {
+                         Err(v @ sea_orm::TryGetError::DbErr(_)) => {
+                             return Err(v);
+                         }
+                         v => v,
+                     };
+                });
+            }
+            ItemType::Skipped => {
+                tokens.extend(quote! {
+                    let #ident = std::default::Default::default();
+                });
+            }
+            ItemType::Nested => {
+                let name = ident.unraw().to_string();
+                tokens.extend(quote! {
+                    let #ident = match sea_orm::FromQueryResult::from_query_result_nullable(row, &format!("{pre}{}-", #name)) {
```

Comment:
Adding the package name to the prefix makes the whole thing unusable unless you are using it with a `DerivePartialModel` because the base model don't add the entity name in front of the column. An integration test would have caught that easily.

Reply:
0. This is not the package name but a prefix computed based on the location (property name) of the nested struct in the parent struct. 
1. There are integration tests.
2. `nested` only makes sense when used with `DerivePartialModel`. 
3. (I have specified this in the comments explicitly as well) This is required to avoid name collisions.

Reply:
Ah, this is about using a struct with `nested` in it in `into_model::<..>()`, I guess? Yup, that is most likely going to break, but due to the name collision thing above there is not really anything I can do about it. The relationship between `FromQueryResult` and `DerivePartialModel` is weird and they should arguably be mutually exclusive, but that would involve copying the entire code from `FromQueryResult` into `DerivePartialModel`. That would also mean doing things like the `skip` I proposed elsewhere twice: